packages/checkout-storefront/src/lib/auth/useSaleorAuthClient.ts

```diff
@@ -0,0 +1,39 @@
+import { SaleorAuthClient, SaleorAuthClientProps } from "./SaleorAuthClient";
+import { useEffect, useMemo, useState } from "react";
+
+export interface UseSaleorAuthClient {
+  saleorAuthClient: SaleorAuthClient;
+  isAuthenticating: boolean;
+}
+
+export const useSaleorAuthClient = ({
```

Comment:
I think there could be a temptation to call this hook in multiple components – and that won't work 😱 
I think we can try making the intention here clearer by renaming it to something like:
```suggestion
/**
 * This hook should be used only once per application.
 * @see Use {@link useSaleorAuthContext} for accessing the auth state.
 */
export const useCreateNewSaleorAuthClient = ({
```

Reply:
Thought making it clear in the docs would be sufficient but sure 👍

Reply:
We could also add a check if a client instance was created inside the project:

```tsx
let clientCreated = true;

export const useSaleorAuthClient = () => {
  // ...
}

```

Reply:
That wouldn't work correctly in many cases.